### DISCUSSION TITLE
raspbery pi 4

### DISCUSSION THREAD

**Discussion by kolosokev** [bol-van/zapret#263] [General]
Добрый день помогите установить на rasbery pi 4 с установленным openwrt 

при выполнении ./install_easy.sh  выдаёт 
build tools not found press enter to continue 

,при выполнении /tmp/zapret/install_bin.sh выдаёт 
x86_64 is NOT OK
x86 is NOT OK
aarch64 is NOT OK
arm is NOT OK
mips64r2-msb is NOT OK
mips32r1-lsb is NOT OK
mips32r1-msb is NOT OK
ppc is NOT OK
no compatible binaries found

но при этом выполняя  uname -m выдаёт 
aarch64

**Comment by bol-van** [comment]
бинарики из  binaries стартуют ?
какая версия   openwrt ?

**Reply by kolosokev** [reply]
<html>
<body>

Hostname | OpenWrt
-- | --
Model | Raspberry Pi 4 Model B Rev 1.1
Architecture | ARMv8 Processor rev 3
Target Platform | bcm27xx/bcm2711
Firmware Version | OpenWrt 23.05.4 r24012-d8dd03c46f / LuCI openwrt-23.05 branch git-24.086.45142-09d5a38
Kernel Version | 5.15.162

не уверен что правильно запускаю но вот что получается
/tmp/zapret# /tmp/zapret/binaries/aarch64/ip2net
Segmentation fault
/tmp/zapret# /tmp/zapret/binaries/aarch64/mdig
Segmentation fault
/tmp/zapret/binaries/aarch64/tpws
Need port number
/tmp/zapret# /tmp/zapret/binaries/aarch64/nfqws
 --debug=0|1
 --qnum=<nfqueue_number> и дальше куча всёго в том же роде , я так понимаю проблема как раз в этом Segmentation fault

</body>
</html>

**Comment by bol-van** [comment]
нужен результат cat /proc/cpuinfo
и еще надо попробовать распаковать upx и запустить без него

**Reply by kolosokev** [reply]
processor       : 0
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 1
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 2
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 3
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

Hardware        : BCM2835
Revision        : c03111
Serial          : 10000000c84d5cab
Model           : Raspberry Pi 4 Model B Rev 1.1

если это поможет могу дать полный доступ к устройству

**Reply by bol-van** [reply]
увы тут лички нет
что такое upx и как его распаковывать есть понятие ?

**Reply by kolosokev** [reply]
распаковал бинари ( увеличились в размере примерно раза в два) , и всё пошло , ./install_easy.sh запустился без проблем
вникаю в настройки что куда и как , спасибо за помощь

**Comment by bol-van** [comment]
давно не обновлял ip2net и mdig
вероятно баг в более  старом  upx
попробуйте их запаковать последним

**Comment by bol-van** [comment]
закомитил все бинарики с новым upx
проверяйте

**Comment by kolosokev** [comment]
ух ты всё получилось ютуб летает , может кому будет интересно для raspbery pi4 firewall зона лан включить маскарадинг и ещё добавить правило для трафика на всякий случай поднять в самый верх правил да у меня только один интерфейс lan как в других случаях не знаю 

название udp433(не принципиально )
протокол UDP
зона источника любая
адрес источника пусто
порт источника пусто
зона назначения любая
адрес назначения любой
порт назначения 443
действие отклонить
